test/unit/gateways/nuvei_test.rb

```diff
@@ -0,0 +1,115 @@
+require 'test_helper'
+
+class NuveiTest < Test::Unit::TestCase
+  include CommStub
+
+  def setup
+    @gateway = NuveiGateway.new(
+      merchant_id: 'SOMECREDENTIAL',
+      merchant_site_id: 'SOMECREDENTIAL',
+      secret_key: 'SOMECREDENTIAL'
+    )
+    @credit_card = credit_card
+    @amount = 100
+
+    @options = {
+      email: 'test@gmail.com',
+      billing_address: address.merge(name: 'Cure Tester'),
+      ip_address: '127.0.0.1'
+    }
+  end
+
+  def supported_card_types
+    assert_equal %i(visa master american_express discover union_pay), NuveiGateway.supported_cardtypes
+  end
+
+  def test_supported_countries
+    assert_equal %w(US CA IN NZ GB AU US), NuveiGateway.supported_countries
+  end
+
+  def build_request_authenticate_url
+    action = :authenticate
+    assert_equal @gateway.send(:url, action), "#{@gateway.test_url}/getSessionToken"
+  end
+
+  def test_scrub
+    assert @gateway.supports_scrubbing?
+    assert_equal @gateway.scrub(pre_scrubbed), post_scrubbed
+  end
+
+  def test_successful_authorize
```

Comment:
🔴 Warning 🔴 

Even when this test is working, is almost a 'false positive', you are using the same stub for the token request and the authorize, is working basically because the token request just checks for the presense of the `sessionToken` attribute.